youtube_dl/extractor/bbc.py

```diff
@@ -912,23 +938,16 @@ def _real_extract(self, url):
                     # http://www.bbc.com/news/world-us-canada-34473351)
                     playlist_object = settings.get('playlistObject', {})
                     if playlist_object:
-                        items = playlist_object.get('items')
-                        if items and isinstance(items, list):
-                            title = playlist_object['title']
-                            description = playlist_object.get('summary')
-                            duration = int_or_none(items[0].get('duration'))
-                            programme_id = items[0].get('vpid')
-                            formats, subtitles = self._download_media_selector(programme_id)
-                            self._sort_formats(formats)
-                            entries.append({
-                                'id': programme_id,
-                                'title': title,
-                                'description': description,
-                                'timestamp': timestamp,
-                                'duration': duration,
-                                'formats': formats,
-                                'subtitles': subtitles,
-                            })
+                        entry = self._extract_from_playlist_object(playlist_object)
+                        programme_id = entry.get('id')
+                        formats, subtitles = self._download_media_selector(programme_id)
+                        self._sort_formats(formats)
+                        entry.update({
+                            'timestamp': timestamp,
+                            'formats': formats,
+                            'subtitles': subtitles,
+                        })
+                        entries.append(entry)
```

Comment:
These looks like candidates for generalization and extracting into a separate method.

Reply:
I created `_get_playlist_entry(entry) -> entry` method to encapsulate the key operations with the benefit of returning `None` when the given `entry` has no `'id'` and checking that result.